examples/common.cpp

```diff
@@ -209,6 +234,14 @@ bool gpt_params_parse(int argc, char ** argv, gpt_params & params) {
         exit(1);
     }
 
+    // Clip if not a valid number of threads
+    if (params.n_threads <= 0) {
+        std::cerr << "\nWARNING: Using number of physical cores as the default number of threads.\n\
+If your chipset has efficient/performance cores, use the number of performance cores instead.\n" << std::endl;
+        int32_t physical_cores = get_num_physical_cores();
+        params.n_threads = std::max(1, physical_cores);
+    }
+
```

Comment:
```suggestion
```
By making a reasonable guess in `get_num_physical_cores` (or even just by returning 1 instead of -1) and calling your function on line 18 of examples/common.h (see below), none of this is necessary.